packages/devtools/lib/src/timeline/frames_bar_chart.dart

```diff
@@ -97,28 +97,78 @@ class PlotlyDivGraph extends CoreElement {
     }
   }
 
+  Selection currentSelection = null;
+
   void _plotlyClick(DataEvent data) {
     final int xPosition = data.points[0].x;
-    if (_frames.containsKey(xPosition)) {
-      final TimelineFrame timelineFrame = _frames[xPosition];
-      framesBarChart.setSelected(timelineFrame);
+
+    final List<int> pointNumbers = [];
+    final List<int> xValues = [];
+    final List<num> yValues = [];
+    final List<int> traces = [];
+
+    for (Point pt in data.points) {
+      // Don't allow selecting an already selected bar.
```

Comment:
what happens when an already selected bar is selected? 
Should we bail out of the click logic early if the selection didn't change or is there a reason we can't?

Reply:
This comment was in the wrong place.  However, there is a quick bailout  !currentSelection.isSelected(newSelection)) I can't bailout until I have the SelectTrace constructed.  See:

    // Don't allow selecting an already selected bar.  If this bar isn't
    // currently selected then select the bar clicked.
    if (newSelection.length == 2 &&
        !currentSelection.isSelected(newSelection)) {
      currentSelection.select(newSelection);